src/Client.php

```diff
@@ -298,10 +298,21 @@ protected function paramsToQueryString(array $params = [])
             if (is_array($value)) {
                 // recursion case
 
-                $result_string = $this->paramsToQueryString($value);
-                if (!empty($result_string)) {
-                    $query_string[] = $result_string;
+                if ( empty( $value ) ) {
+                    $query_string[] = $name;
+                    continue;
                 }
+
+                foreach ( $value as $key => $item ) {
```

Comment:
Thanks for the improvement PR @martin-rueegg 
What is the reason not to use recursion here?

Reply:
As shown in the description of the PR, the only way to provide multiple countries, is and array like this:

```[ 'country' => ['AD', 'CH','UK'] ]```

which should result in something like this:

```country=AD&country=CH&country=UK```

Hence, in [line 289](https://github.com/Aternus/geonames-client/blob/b01e63d2bac6f605f0caa365bed618a278ce017d/src/Client.php#L298) I check for an array and subsequently I have to walk through the array in order to add all the values to the resulting array `$query_string[]`.

Reply:
Alright. So it seems this function needs a bit of refactoring to so that we're not repeating ourselves with `urlencode` in each case. Also, please remove the comments about recursion (recursive case, base case).